### PR TITLE
NAS-115257 / 22.12 / NAS-115257: Refactor AssociatedTargetForm and AuthorizedAccessForm

### PR DESCRIPTION
**Pull request by undsoft**
Testing:
Shares -> iSCSI -> Configure.
Test forms on Authorized Access and Associated Targets.

You may need to create additional entries on other tabs.

### CODE REVIEW

File: src/app/pages/sharing/iscsi/associated-target/associated-target-form/associated-target-form.component.ts [hunk 1]
```diff
@@ -1,130 +1,95 @@
...
+      : this.translate.instant('Edit Authorized Target');
+  }
 
-  fieldConfig: FieldConfig[];
+  form = this.formBuilder.group({
+    target: [null as number, Validators.required],
+    lunid: [null as number, [
+      Validators.min(0),
+      Validators.max(1023),
+      Validators.pattern(/^(0|[1-9]\d*)$/),
```
Comment (AlexKarpov98) [inline src/app/pages/sharing/iscsi/associated-target/associated-target-form/associated-target-form.component.ts:36]: `Validators.pattern` for number is not needed.

File: src/app/pages/sharing/iscsi/authorized-access/authorized-access-form/authorized-access-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      ),
+      this.validatorService.withMessage(
+        matchOtherValidator('peersecret_confirm'),
+        {
+          message: this.translate.instant('Secret and confirmation should match'),
+          forProperty: 'matchOther',
+        },
+      ),
+    ]],
+    peersecret_confirm: [''],
```
Comment (AlexKarpov98) [inline src/app/pages/sharing/iscsi/authorized-access/authorized-access-form/authorized-access-form.component.ts:74]: I think It's better to add validation like `peersecret`
```
      this.validatorService.validateOnCondition(
        () => this.isPeerUserSet(),
        Validators.required,
      ),
```

**Review by AlexKarpov98 — APPROVED** [review]
LGTM! 👍
Screenshot_35
Screenshot_36

File: src/app/pages/sharing/iscsi/authorized-access/authorized-access-form/authorized-access-form.component.ts (hunk 2 above)
Comment (undsoft) [reply src/app/pages/sharing/iscsi/authorized-access/authorized-access-form/authorized-access-form.component.ts:74]: Not sure I understand this point.

Comment (undsoft) [reply src/app/pages/sharing/iscsi/authorized-access/authorized-access-form/authorized-access-form.component.ts:74]: I think I understand. This field will be validated by `matchOtherValidator('peersecret_confirm'),` above, which in turn will be required.

**Pull request merged**